backend/windmill-worker/src/worker_flow.rs

```diff
@@ -1495,12 +1495,35 @@ pub async fn handle_flow(
     worker_dir: &str,
     job_completed_tx: Sender<SendResult>,
 ) -> anyhow::Result<()> {
-    let flow = flow_value
+    let mut flow = flow_value
         .with_context(|| "Unable to parse flow definition")?;
     let status = flow_job
         .parse_flow_status()
         .with_context(|| "Unable to parse flow status")?;
 
+    // Update value.
+    if matches!(flow_job.job_kind, JobKind::Flow) {
```

Comment:
Check if this mutation is really necessary; same_worker can be taken from job and skipping the preprocessor can be done later on based on this same next condition